### PR TITLE
Update ReadableStreamTee to allow cloning for only the second branch

### PR DESCRIPTION
**Pull request by domenic**
This is part of fixing https://github.com/yutakahirano/fetch-with-streams/issues/67.

/cc @youennf @yutakahirano. Because of how I changed this (just modifying the meaning of the second arg being true) this should require no subsequent changes to Fetch.

### CODE REVIEW

**Comment by youennf** [comment]
That seems ok as a minimal change.
I would like to understand though why a direct clone method is not possible/appropriate.
The other naive question I have is whether we can do so something better in the case the source is mandated to be a byte stream source.

File: reference-implementation/lib/readable-stream.js [hunk 1]
```diff
@@ -488,23 +488,20 @@ function create_ReadableStreamTeePullFunction() {
...
 
+      const value1 = value;
+      const value2 = value;
+
       // There is no way to access the cloning code right now in the reference implementation.
       // If we add one then we'll need an implementation for StructuredClone.
-
+      // if (teeState.canceled2 === false && cloneForBranch2 === true) {
+      //   value2 = StructuredClone(value2);
+      // }
```
Comment (tyoshino) [inline reference-implementation/lib/readable-stream.js]: Even when the branch 1 has been cancelled, we clone values. I think it's ok and is more consistent, but leaving this comment to make sure you're aware of.

File: index.bs [hunk 2]
```diff
@@ -782,14 +769,11 @@ ReadableStreamTee pull function <var>F</var> is called, it performs the followin
...
-      1. Perform ? ReadableStreamDefaultControllerEnqueue(_branch1_, _value1_).
-    1. If _teeState_.[[canceled2]] is *false*,
-      1. Let _value2_ be _value_.
-      1. If _shouldClone_ is *true*, set _value2_ to ? <a abstract-op>StructuredClone</a>(_value_).
-      1. Perform ? ReadableStreamDefaultControllerEnqueue(_branch2_, _value2_).
+    1. Let _value1_ and _value2_ be _value_.
+    1. If _teeState_.[[canceled2]] is *false* and _cloneForBranch2_ is *true*, set _value2_ to ? <a
+       abstract-op>StructuredClone</a>(_value2_).
+    1. If _teeState_.[[canceled1]] is *false*, perform ? ReadableStreamDefaultControllerEnqueue(_branch1_, _value1_).
+    1. If _teeState_.[[canceled2]] is *false*, perform ? ReadableStreamDefaultControllerEnqueue(_branch1_, _value2_).
```
Comment (tyoshino) [inline index.bs]: _branch2_

File: index.bs [hunk 3]
```diff
@@ -28,11 +28,6 @@ Ignored Vars: e
 </pre>
 
 <pre class="anchors">
-urlPrefix: https://html.spec.whatwg.org/multipage/; spec: HTML
-    text: StructuredClone; urlPrefix: infrastructure.html; type: abstract-op
-    text: Transfer; url: infrastructure.html#transfer-abstract-op; type: abstract-op
-    text: WebSocket; url: comms.html#websocket; type: interface
-    text: transferable objects; url: infrastructure.html; type: dfn
```
Comment (tyoshino) [inline index.bs]: what made these unnecessary?

**Comment by domenic** [comment]
> I would like to understand though why a direct clone method is not possible/appropriate.

To read the data from a readable stream, you must necessarily remove it from the stream's internal queue. So there is no real way to create a clone; the very act of reading is destructive.

File: reference-implementation/lib/readable-stream.js (hunk 1 above)
Comment (domenic) [inline reference-implementation/lib/readable-stream.js]: Ah, that is interesting; thanks for pointing it out. It is probably better this way, yeah.

File: index.bs (hunk 3 above)
Comment (domenic) [inline index.bs]: Recent changes to the HTML spec to export these terms automatically instead of requiring manual import.

Comment (domenic) [inline index.bs]: I pulled this and some other changes that are unrelated to teeing out into https://github.com/whatwg/streams/commit/0dc890351d298e288161c991582d998a9d5e5f51 (committed to master)

File: index.bs (hunk 2 above)
Comment (domenic) [inline index.bs]: Thanks, fixed

**Comment by tyoshino** [comment]
3a95925 lgtm :)

**Comment by yutakahirano** [comment]
lgtm
